core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalCheckpointThread.java

```diff
@@ -127,8 +129,10 @@ public UfsJournalCheckpointThread(Master master, UfsJournal journal, long startS
     mCheckpointPeriodEntries =
         Configuration.getInt(PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES);
     mJournalSinks = journalSinks;
+    setName(String.format("ufs-checkpoint-thread-%s", mMaster.getName()));
+    // Override the exception handler with more handling
     setUncaughtExceptionHandler((thread, t) -> {
-      mThrowable = t;
+      setError(t);
```

Comment:
Actually this is a different UncaughtExceptionHandler from the parent, so I should probably change the logic to calling an extensible method as lambda.

Reply:
changed